packages/nimble-components/src/icon-base/tests/icon-metadata.ts

```diff
@@ -100,14 +109,29 @@ export const iconMetadata: {
     IconBookMagnifyingGlass: {
         tags: ['analyze']
     },
-    IconCalendar: {
+    IconCalendarMonth: {
         tags: ['schedule']
     },
+    IconCalendarClockSchedulingAssistant: {
```

Comment:
Can you make sure the names are following the naming guidelines?
https://github.com/ni/nimble/blob/main/packages/nimble-tokens/CONTRIBUTING.md#icon-naming

High-level concerns:

- These names don't seem like what I expect from font awesome, but I did not search so may help to reference if used
- Names like `IconCalendarThreeRectanglesAgenda` seem to be mixing both "descriptive names" (calendar three rectangles) and "Metaphors" (agenda)

May be useful to update contributing docs for icon names to encourage linking to the relevant font awesome names in PR description or PR comments.

Reply:
@NIbokeefe wanted these specific names. It seems like whoever decides on the naming scheme is intentionally adding metaphors to the names. He already expressed some concern about our element names not matching the SVG filenames exactly, so I think we need to have a discussion.

Reply:
Throwing up the @fredvisser bat signal 🦇🔦

Reply:
I understand Brandon's concerns, but we need to use the font awesome naming scheme, or this will descend into chaos. I.e.

IconCalendarClockSchedulingAssistant → IconCalendarClock
IconCalendarOneSquareDay → IconCalendarDay
IconCalendarOneHollowSquareToday → IconCalendarToday
IconCalendarThreeRectanglesAgenda → IconCalendarLines
IconCalendarTopRectangleWeek →  IconCalendarWeek
IconCalendar/IconCalendarMonth → IconCalendarDays (we could defer this to avoid a breaking change)

IconArrowURedo → IconArrowRotateRight
IconArrowUUndo → IconArrowRotateLeft

IconLinkRightArrow → IconArrowRight

Reply:
There's some stuff we need to get consensus on.

1. Fred, the names you propose are a mix of descriptions and metaphors, for example `IconCalendarDay` (metaphor) vs `IconCalendarOneSquare` (description). I can see the argument for the metaphor-based name, since there really shouldn't be any other use case/metaphor for that icon, but it's still inconsistent with our naming pattern so far (I think).
2. If `IconLinkRightArrow` is the icon we'll use specifically for indicating links, should we reserve it for that specific purpose and use some other variation of a right-pointing arrow for other contexts? The same question goes for `IconAnchorBoxUpRightArrow`. My concern is that an icon that has established itself to mean one thing in 99% of use cases may be confusing when used in another context 1% of the time.
3. We should pick either "link" or "anchor" and consistently use that term (in icon names, at least).
4. Fred, your recommended names `IconArrowRotateRight` and `IconArrowRotateLeft` are too ambiguous and the former conflicts with an existing icon name. We also have an existing icon `IconArrowURotateLeft` which is not a great name, because (1) the arrow head points up, not to the left, and (2) the arrow makes a clockwise turn, which is not what "rotate left" would imply. I propose we rename the existing icon `IconArrowUUp`, and then our new ones logically follow as `IconArrowULeft` and `IconArrowURight`. This naming scheme assumes we'll never need both "isomers" of the icon, e.g.
![image](https://github.com/ni/nimble/assets/7282195/66b6bdcd-55be-4bb9-bad9-52cee1692ef3)

Reply:
This is the beauty of our [icon naming guidance](https://github.com/ni/nimble/blob/main/packages/nimble-tokens/CONTRIBUTING.md#icon-naming):

>Use [Font Awesome](https://fontawesome.com/) names for icons. If Font Awesome doesn't contain an equivalent icon, minimize the use of metaphors; choose descriptive and unambiguous names instead.

Font Awesome [has](https://fontawesome.com/search?q=calendar&o=r):
- calendar-clock
- calendar-day
- calendar-lines
- calendar-week
- calendar-days

…so those are all good. These aren't metaphors so much as clear descriptions of the icons. I.e. It's not _agenda_ or _schedule_.

And for the link icons, font awesome also [has those](https://fontawesome.com/search?q=link&o=r):
- arrow-up-right-from-square
- arrow-right

…and we shouldn't include link or anchor in the icon name for an arrow. We can expand our icon documentation elsewhere, but this is not the place to bake it in.

<img width="706" alt="Screenshot 2024-07-11 at 2 01 10 PM" src="https://github.com/ni/nimble/assets/1458528/0e8d7e80-0797-43ad-ab1c-271800dae5ab"> 🤦‍♂️

And yeah, for #4 - I like your proposal - 👍